public/app/core/components/AppChrome/MegaMenu/hooks.ts

```diff
@@ -0,0 +1,14 @@
+import { useMemo } from 'react';
+
+import { config } from '@grafana/runtime';
+import { useGetUserPreferencesQuery } from 'app/features/preferences/api';
+
+export const usePinnedItems = () => {
+  const preferences = useGetUserPreferencesQuery();
+  const pinnedItems = useMemo(() => preferences.data?.navbar?.savedItemIds || [], [preferences]);
+
+  if (config.featureToggles.pinNavItems) {
```

Comment:
Nit: Should this `if` be at the start of the function so we don't make any unnecessary queries to the preferences endpoint?

Reply:
Hooks can't be called conditionall, we'd need to add a condition inside the API call itself. I don't think it's a big deal as it only fetches org preferences and the feature toggle is temporary anyways.